### PR TITLE
Fix Problems with non-space spaces in VMC.

### PR DESCRIPTION
**Pull request by BillWhite**
The HTML DOM puts invisible non-space characters in text.  We ignore it here.

I also changed build.xml so that the JavaScript is not minified in a debug build.  This means it's debuggable, though not easily debuggable.

### CODE REVIEW

**Comment by yzhang1991** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[build.xml, line 5 at r1](https://reviewable.io:443/reviews/voltdb/voltdb/5137#-L3aQVW0jisaMAv8ifvL:-L3aQVW1M6gVs_h0s38n:b88ieil) ([raw file](https://github.com/voltdb/voltdb/blob/9247697f163b8e543484863fa40232f47a894469/build.xml#L5)):*
> ```XML
> 	     name="VoltDB" 
> 	     xmlns:if="ant:if"
> 		 xmlns:unless="ant:unless"
> ```

Is this indentation correct?

---

*[build.xml, line 1228 at r1](https://reviewable.io:443/reviews/voltdb/voltdb/5137#-L3aRU-y6Va-kc2lb9uf:-L3aRU-y6Va-kc2lb9ug:b-ng6v8b) ([raw file](https://github.com/voltdb/voltdb/blob/9247697f163b8e543484863fa40232f47a894469/build.xml#L1228)):*
> ```XML
>         <fileset dir="temp/js" includes="**/*.js" />
>         <mapper type="glob" from="*.js" to="*-abc.js"/>
> 	</copy>		
> ```

Can you remove the ending <tab> in this line?

---

*[src/frontend/org/voltdb/dbmonitor/css/layout.css, line 1961 at r1](https://reviewable.io:443/reviews/voltdb/voltdb/5137#-L3aT0uzRWsmtfIaTFYu:-L3aT0uzRWsmtfIaTFYv:b-1zs4yy) ([raw file](https://github.com/voltdb/voltdb/blob/9247697f163b8e543484863fa40232f47a894469/src/frontend/org/voltdb/dbmonitor/css/layout.css#L1961)):*
> ```CSS
>     height: 85%;
>     overflow: auto;
>     white-space: pre;
> ```

After reading Andrew's email, I think for HTML format results, we should use `white-space: pre-wrap;`; and for CSV and monospace format results, we should use `white-space: pre;`.

Using `pre` in HTML format results may result in extra-long rows because it does not wrap at all unless specified in the text. This will be inconsistent with our previous behavior, and, not pretty.

Wrapping text in monospace and CSV files can be troublesome if I want to copy and paste the result for other uses. I vote for removing them in those two modes using `pre`.

**Howver**, the changes I am suggesting in this comment should not happen here. As I examine the code, the style here in the `.resultHtml` will be overridden by the changes you made in `tablesorter`. I will suggest my changes there.

**Reference:** https://developer.mozilla.org/en-US/docs/Web/CSS/white-space

---

*[src/frontend/org/voltdb/dbmonitor/css/tablesorter.css, line 12 at r1](https://reviewable.io:443/reviews/voltdb/voltdb/5137#-L3aXFOYQxPVjmBkgLOP:-L3aXFOZHAvvit9J2Mkw:b-j6eu7u) ([raw file](https://github.com/voltdb/voltdb/blob/9247697f163b8e543484863fa40232f47a894469/src/frontend/org/voltdb/dbmonitor/css/tablesorter.css#L12)):*
> ```CSS
> 	cursor: pointer;
> }
> table.tablesorter td {
> ```

The `tablesorter` class is also used in the "Schema" tab, where we need auto-wrapping. This is important because some tables there can have extra-long lines. For example, the explanation string for a query plan that accesses a table with 20 columns with complex filters, view on joins often have that. So this may better be `pre-wrap`

---

*[src/frontend/org/voltdb/dbmonitor/css/tablesorter.css, line 17 at r1](https://reviewable.io:443/reviews/voltdb/voltdb/5137#-L3aj0bNBbdqvzy7-INH:-L3aj0bNBbdqvzy7-INI:b-ptyjb6) ([raw file](https://github.com/voltdb/voltdb/blob/9247697f163b8e543484863fa40232f47a894469/src/frontend/org/voltdb/dbmonitor/css/tablesorter.css#L17)):*
> ```CSS
> table.tablesorter tbody tr.odd td {
> 	background-color:#f9f9f9;
> 	white-space:pre;
> ```

Where do you find this needs to be changed? My understanding is that this style will be inherited from `table.tablesorter td`. If you found something that proves it does not inherit, please point it out.

---

*[src/frontend/org/voltdb/dbmonitor/js/voltdb.queryrender.js, line 138 at r1](https://reviewable.io:443/reviews/voltdb/voltdb/5137#-L3ajdyFN5nn3o4LyOPT:-L3ajdyFN5nn3o4LyOPU:bfjrfqc) ([raw file](https://github.com/voltdb/voltdb/blob/9247697f163b8e543484863fa40232f47a894469/src/frontend/org/voltdb/dbmonitor/js/voltdb.queryrender.js#L138)):*
> ```JavaScript
>                 replStr = stringBank[QuotedStringNonceLiteral + nonceNum];
>                 src = src.replace(QuotedStringNonceLiteral + nonceNum,
>                             	  replStr);
> ```

Is this a <tab>? 
Can you explain this part a little bit?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/voltdb/voltdb/5137)*

**Comment by yzhang1991** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[build.xml, line 5 at r1](https://reviewable.io:443/reviews/voltdb/voltdb/5137#-L3aQVW0jisaMAv8ifvL:-L3akEJt0jDywkj44do4:bnq2xh9) ([raw file](https://github.com/voltdb/voltdb/blob/9247697f163b8e543484863fa40232f47a894469/build.xml#L5)):*
<details><summary><i>Previously, yzhang1991 (Yiqun Zhang) wrote…</i></summary><blockquote>

Is this indentation correct?
</blockquote></details>

You are using tabs here,  can you replace with spaces?

---

*[build.xml, line 1228 at r1](https://reviewable.io:443/reviews/voltdb/voltdb/5137#-L3aRU-y6Va-kc2lb9uf:-L3akLfrCJ-cjeyaZyZR:b8e40v8) ([raw file](https://github.com/voltdb/voltdb/blob/9247697f163b8e543484863fa40232f47a894469/build.xml#L1228)):*
<details><summary><i>Previously, yzhang1991 (Yiqun Zhang) wrote…</i></summary><blockquote>

Can you remove the ending <tab> in this line?
</blockquote></details>

Same here, can you replace the tabs near this area with spaces?

---

*[src/frontend/org/voltdb/dbmonitor/js/voltdb.queryrender.js, line 138 at r1](https://reviewable.io:443/reviews/voltdb/voltdb/5137#-L3ajdyFN5nn3o4LyOPT:-L3akn_IoNdvO0NFKi3C:b29ez62) ([raw file](https://github.com/voltdb/voltdb/blob/9247697f163b8e543484863fa40232f47a894469/src/frontend/org/voltdb/dbmonitor/js/voltdb.queryrender.js#L138)):*
<details><summary><i>Previously, yzhang1991 (Yiqun Zhang) wrote…</i></summary><blockquote>

Is this a <tab>? 
Can you explain this part a little bit?
</blockquote></details>

There is a tab here, I don't know why our licensecheck do not cover this, but please fix it.

It would be really helpful if you could explain this part of the change a little bit more.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/voltdb/voltdb/5137)*

**Comment by yzhang1991** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[src/frontend/org/voltdb/dbmonitor/js/voltdb.queryrender.js, line 138 at r1](https://reviewable.io:443/reviews/voltdb/voltdb/5137#-L3ajdyFN5nn3o4LyOPT:-L3aq9XB2pkaivLMYEoL:bg7aqmr) ([raw file](https://github.com/voltdb/voltdb/blob/9247697f163b8e543484863fa40232f47a894469/src/frontend/org/voltdb/dbmonitor/js/voltdb.queryrender.js#L138)):*
<details><summary><i>Previously, yzhang1991 (Yiqun Zhang) wrote…</i></summary><blockquote>

There is a tab here, I don't know why our licensecheck do not cover this, but please fix it.

It would be really helpful if you could explain this part of the change a little bit more.
</blockquote></details>

Oh never mind, I was thinking about the wrong thing

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/voltdb/voltdb/5137)*

File: .gitignore [hunk 1]
```diff
@@ -19,6 +19,7 @@
 *runapp*.py
 *.log
 *.iml
+/build.local
```
Comment (BillWhite) [inline .gitignore]: This is a buildtools.py thing.

File: build.xml [hunk 2]
```diff
@@ -1221,12 +1221,12 @@ JAVA COMPILATION
     <copy todir="temp/js" overwrite="yes">
         <fileset dir="src/frontend/org/voltdb/dbmonitor/js"/>
     </copy>
-	<!-- Do not compress javascript unless this is a release build. -->
-	<copy todir="temp/js" overwrite="yes" unless:set='isRelease'>
+        <!-- Do not compress javascript unless this is a release build. -->
+        <copy todir="temp/js" overwrite="yes" unless:set='isRelease'>
```
Comment (yzhang1991) [inline build.xml]: Sorry to bother you with this, but the indentation here is still not correct.

File: build.xml [hunk 3]
```diff
@@ -1269,11 +1269,11 @@ JAVA COMPILATION
         </fileset>
     </copy>
     <copy flatten='false' todir='${build.prod.dir}'>
-		<fileset dir="${src.gpl.dir}">
+                <fileset dir="${src.gpl.dir}">
```
Comment (yzhang1991) [inline build.xml]: Indentation

**Pull request merged**